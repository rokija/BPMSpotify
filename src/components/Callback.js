import { Component } from 'react';
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { getAuth, validateCallbackResult, getUserData } from '../actions/authActions';

class Callback extends Component {

    constructor() {
        super();
    }

    componentWillMount() {
        const {dispatch} = this.props;

        dispatch(validateCallbackResult(location.hash));
    }

    componentWillReceiveProps(nextProps) {
        const {authReducer, dispatch} = nextProps;
        dispatch(getUserData());
        if(!authReducer || Object.keys(authReducer).length === 0 && authReducer.constructor === Object){
             // console.log("NOTT")
            // return null;
        }

        if(authReducer.isLogged) {
            this.context.router.history.push("/");
        } else {
            // console.log("errrr")
            // this.context.router.push("/errorPage");
        }
    }

    render() {
        return null;
    }
}

Callback.propTypes = {
    dispatch: PropTypes.func,
    authReducer: PropTypes.object,
    userLoginState: PropTypes.object
};

Callback.contextTypes = {
    router: PropTypes.object
};

function mapStateToProps (state) {
    const {authReducer, userReducer} = state;
    return {
        authReducer,
        userReducer
    };
}

function mapDispatchToProps (dispatch) {
    return {
        dispatch,
        actions: bindActionCreators({getAuth, validateCallbackResult, getUserData}, dispatch)
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Callback);
