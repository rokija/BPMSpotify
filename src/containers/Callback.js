import { Component } from 'react';
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { getAuth, validateCallbackResult } from '../actions/authActions';

export class Callback extends Component {

    componentWillMount() {
        const {dispatch} = this.props;
        dispatch(validateCallbackResult(location.hash));
    }

    componentWillReceiveProps(nextProps) {
        const {authReducer} = nextProps;
        if(authReducer.isLogged) {
            this.context.router.history.push("/search");
            return true;
        }
    }

    render() {
        return null;
    }
}

Callback.propTypes = {
    dispatch: PropTypes.func,
    authReducer: PropTypes.object,
    userLoginState: PropTypes.object
};

Callback.contextTypes = {
    router: PropTypes.object
};

export function mapStateToProps (state) {
    const {authReducer} = state;
    return {
        authReducer
    };
}

export function mapDispatchToProps (dispatch) {
    return {
        dispatch,
        actions: bindActionCreators({getAuth, validateCallbackResult}, dispatch)
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Callback);
