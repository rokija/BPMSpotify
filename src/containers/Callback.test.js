import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Callback, mapStateToProps, mapDispatchToProps } from './Callback';

const context = { router: { history: [] }  };

const minProps = {
    dispatch: jest.fn(),
    authReducer: { isLogged: true }
};

describe('Callback', () => {
    const wrapper = shallow(<Callback {...minProps} />, { context });
    const wrapperInstance = wrapper.instance();

    it('renders properly', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('calls componentWillReceiveProps', () => {
        expect(wrapperInstance.componentWillReceiveProps(minProps)).toEqual(true);
    });

    it('calls mapStateToProps', () => {
        expect(mapStateToProps({authReducer: { isLogged: true }})).toEqual({"authReducer": {"isLogged": true}});
    });

    it('calls typeof mapDispatchToProps', () => {
        expect(typeof (mapDispatchToProps())).toEqual("object");
    });

    it('matches snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});