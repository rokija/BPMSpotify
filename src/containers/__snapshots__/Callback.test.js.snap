// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Callback matches snapshot 1`] = `""`;
